### PR TITLE
Add method to search local posts by title

### PR DESCRIPTION
**Pull request by tonyr59h**
This is similar to the [MediaStore's search method](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java#L419). wordpress-mobile/WordPress-Android#62 will be using this to search the post list.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/PostStore.java [hunk 1]
```diff
@@ -275,6 +275,14 @@ public PostModel getPostByLocalPostId(long localId) {
         }
     }
 
+    public List<PostModel> searchPostTitles(SiteModel site, String searchTerm) {
```
Comment (catehstn) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/PostStore.java]: can there be unit tests for these methods?

Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/PostStore.java]: Can do 👍

**Comment by maxme** [comment]
We should wait before merging this.

1. Are we adding local search in the app? I personally think this is confusing and we should only have online search for .com/jetpack users.
2. If yes, are we searching by matching title only?

**Comment by tonyr59h** [comment]
I think searching locally makes sense. If there's a bunch of posts cached locally I'd expect (as a user) that the search feature would be able to filter them without a network connection. This is something we need more input for it seems. cc @mattmiklic 

[The WPAndroid branch](https://github.com/wordpress-mobile/WordPress-Android/tree/issue/62-search-posts-non-searchable) is currently hiding the search field for self-hosted sites. I'll update it to check for Jetpack as well.

**Comment by tonyr59h** [comment]
Closing this in favor of an online search.

**Pull request closed without merging**